src/util/tile-error-collection.js

```diff
@@ -0,0 +1,80 @@
+var Backbone = require('backbone');
+var $ = require('jquery');
+var getValue = require('./get-object-value');
+var tileErrorImage = require('../../themes/img/tile-error.svg');
+
+var TILE_ERROR_IMAGE = 'data:image/svg+xml;base64,' + window.btoa(tileErrorImage);
+
+var TileErrorCollection = Backbone.Collection.extend({
+  initialize: function () {
+    this.running = false;
+    this.queue = new Backbone.Collection();
+
+    this.on('add', this._onAdd, this);
+  },
+
+  resetErrorTiles: function () {
+    this.running = false;
+    this.queue.reset();
+
+    var models = this.filter(function (model) {
+      this._deletedNode(model.get('tileDomNode'));
+    }.bind(this));
+
+    this.reset(models);
+  },
+
+  getError: function () {
+    var model = this.find(function (model) {
+      return model.has('error');
+    });
+
+    return model && model.get('error');
+  },
+
+  _onAdd: function (model) {
+    var isInQueue = this.queue.findWhere({ url: model.get('url') });
+    if (!isInQueue) this.queue.add(model);
+
+    model.get('tileDomNode').src = TILE_ERROR_IMAGE;
+
+    if (!this.running) {
+      this.running = true;
+      this._getTileErrors();
+    }
+  },
+
+  _getTileErrors: function () {
+    if (!this.running) return;
+
+    var model = this.queue.find(function (model) {
```

Comment:
Why not just remove the model from the queue to ensure is not checked again?

Reply:
Because it could fail when loading in the `img`, and then the ajax request be successful, but fail again, at this could go on forever, so I have to make sure we just make one ajax call.

Reply:
We're always setting `checked` to `true` after the request is done and then we filter those "tiles"... isn't that the same as removing it from the queue so that it doesn't get "processed" again?

Reply:
Yes, but then in `_onAdd` we check it's not in the queue so we don't add it again

Reply:
Why would a tile be added to this collection more than once?

Reply:
Because how Maps API works, when requesting a tile, it can timeout in 2 places, in an "internal queue" they have or in the database, if it fails in the database it will always fail, but there is a case when it will timeout in the queue, the database will finish and cache the tile, and the next request (the AJAX one) will work, that's ok in most of the cases, because the cache will keep alive, but in local (or maybe in onpremise?) there is a configuration which resets the cache on hit, so the first request for loading the image will fail, the ajax one will return a 200, and the next try to load the image (via src) will fail again, because the cache has been removed, that leads to an infinite loop of adding to the queue, status 200, fail, adding to the queue, status 200, fail, so the solution I found was to mark them as "checked" so we don't add them multiple times when this happens.

Reply:
Thanks for the explanation. Still, isn't `tileerror` (or `tile.onerror` for Google Maps) only called once for each tile?

Reply:
No, because when the ajax call returns a `200`, I update the image `src` with the tile url, which could fail again, triggering another `tileeror`

Reply:
Ok, that explains a lot of things! I thought `tileerror` was only being triggered when Leaflet tried to load each of the tiles, but I wasn't expecting it'd be triggered when we change the `src` of the tile afterwards.